### PR TITLE
Refactor `Parallel_Orbitals`

### PR DESCRIPTION
**Pull request by maki49**
- add class `Parallel_2D` for 2D-block-division of an **abstract matrix with arbitrary size** (with UT)
- 2d-variables and functions associated with wavefunction/atoms are still in `Parallel_Orbitals`, others are moved into `Parallel_2D`
- extract some interfaces from `ORB_control` to `Parallel_2D` and `Parallel_Orbitals`

---
why refactor: 2D-block-division of a new matrix X (nocc*nvirt) will be implemented  (#2460 )

### CODE REVIEW

**Comment by maki49** [comment]
I think it will be developer-friendly to rename the index-mapping array `trace_loc_row` as `global2local_row`, and `row_set` as `local2global_row`. 
@dyzheng If you agree, I will do this in another PR (since tens of files will be changed) after this PR is merged.

File: source/module_basis/module_ao/parallel_orbitals.h [hunk 1]
```diff
@@ -1,45 +1,13 @@
...
-	int col_pos;
-    int row_pos;
-    
-	int row_b;  //row block size
-    int col_b;  //column block size
-};
+#pragma once
+#include "parallel_2d.h"
 
 /// These stucture packs the information of 2D-block-cyclic
```
Comment (dyzheng) [inline source/module_basis/module_ao/parallel_orbitals.h]:
```suggestion
/// This class packs the information of 2D-block-cyclic for LCAO code.
```

File: source/module_basis/module_ao/parallel_2d.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#pragma once
+#include "module_base/global_function.h"
+#include "module_base/global_variable.h"
+
+#ifdef __MPI
+#include <mpi.h>
+#endif
+
+/// @brief  This structure packs the basic information of
```
Comment (dyzheng) [inline source/module_basis/module_ao/parallel_2d.h]:
```suggestion
/// @brief  This class packs the basic information of
```

File: source/module_basis/module_ao/parallel_orbitals.cpp [hunk 3]
```diff
@@ -122,4 +114,72 @@ int Parallel_Orbitals::get_row_size(int iat) const
...
     throw std::string("error in get_col_size(iat)");
-}
\ No newline at end of file
+}
+
+#ifdef __MPI
+void Parallel_Orbitals::set_desc_wfc_Eij(const int& nbasis, const int& nbands, const int& lld)
+{
+    ModuleBase::TITLE("Parallel_2D", "set_desc_wfc_Eij");
+    assert(this->comm_2D != MPI_COMM_NULL);
```
Comment (dyzheng) [inline source/module_basis/module_ao/parallel_orbitals.cpp:124]:
```suggestion
#ifdef __DEBUG
    assert(this->comm_2D != MPI_COMM_NULL);
```

File: source/module_basis/module_ao/parallel_orbitals.cpp [hunk 4]
```diff
@@ -122,4 +114,72 @@ int Parallel_Orbitals::get_row_size(int iat) const
...
+    assert(nbasis > 0 && nbands > 0 && lld > 0);
+    assert(this->nb > 0 && this->dim0 > 0 && this->dim1 > 0);
```
Comment (dyzheng) [inline source/module_basis/module_ao/parallel_orbitals.cpp:126]:
```suggestion
    assert(this->nb > 0 && this->dim0 > 0 && this->dim1 > 0);
#endif
```

File: source/module_basis/module_ao/parallel_orbitals.h [hunk 5]
```diff
@@ -1,45 +1,13 @@
...
-	int col_num;
-    int row_num;
-    
-	int col_pos;
-    int row_pos;
-    
-	int row_b;  //row block size
-    int col_b;  //column block size
-};
+#pragma once
```
Comment (dyzheng) [inline source/module_basis/module_ao/parallel_orbitals.h]: please use ifndef but not pragma once for consistent code style

File: source/module_basis/module_ao/test/parallel_2d_test.cpp [hunk 6]
```diff
@@ -0,0 +1,131 @@
+#include "gtest/gtest.h"
+#include "../parallel_2d.h"
+class test_para2d : public testing::Test
```
Comment (dyzheng) [inline source/module_basis/module_ao/test/parallel_2d_test.cpp:36]: Please add annotations for your UTs.

**Review by dyzheng — COMMENTED** [review]
Changes in this PR are good.

**Review by dyzheng — APPROVED** [review]

**Pull request merged**